gallery/tutorials/aspire_introduction.py

```diff
@@ -0,0 +1,675 @@
+# Anaconda Python, by following the instructions in the README.  `The
+# instructions for developers is the most comprehensive
+# <https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/master/README.md#for-developers>`_.
+# Windows is provided, but generally Linux and OSX are recommended,
+# with Linux being the most diversely tested platform.
+#
+
+# %%
+# Princeton Research Computing
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# ASPIRE requires some resources to run, so if you wouldn't run
+# typical data science codes on your machine (a netbook for example),
+# you may use Tiger/Adroit/Della at Princeton or another cluster.
+# After logging into Tiger, ``module load anaconda3/2020.7`` and
+# continue to follow the anaconda instructions for developers in the
+# link above.  Those instructions should create a working environment
+# for tinkering with ASPIRE code found in this notebook.
+
+# %%
+# Imports
+# ^^^^^^^
+# First we import some typical scientific computing packages.
+# Along the way we will import relevant components from ``aspire``.
+# Users may also import ``aspire`` once as a top level package.
+
+import logging
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import aspire
+from aspire.image import Image
+
+# %%
+# API Primitives
+# --------------
+# The ASPIRE framework is a collection of modules containing
+# interoperable extensible components.  Underlying the more
+# sophisticated components and algorithms are some core data
+# structures.  Sophisticated components are designed to interoperate
+# by exchanging, consuming, or producing these basic structures.  The
+# most common structures encountered when starting out are:
+
+# %%
+# .. list-table:: Core API Components
+#    :header-rows: 1
+#
+#    * - Component
+#      - Description
+#    * - ``Image``
+#      - Utility class for stacks of 2D arrays.
+#    * - ``Volume``
+#      - Utility class for stacks of 3D arrays.
+#    * - ``Rotations``
+#      - Utility class for stacks of 3D rotations.
+#    * - ``Filter``
+#      - Constructs and applies Image filters.
+#    * - ``Basis``
+#      - Basis conversions and operations.
+#    * - ``Source``
+#      - Produces primitive components. ``ImageSource`` produces ``Images``.
+
+
+# %%
+# ``Image`` Class
+# ---------------
+#
+# The `Image
+# <https://computationalcryoem.github.io/ASPIRE-Python/aspire.image.html#aspire.image.image.Image>`_
+# class is a thin wrapper over Numpy arrays for a stack containing 1
+# or more images (2D data).  In this notebook we won't be working
+# directly with the ``Image`` class a lot, but it will be one of the
+# fundemental structures behind the scenes.  A lot of ASPIRE code
+# passes around ``Image`` and ``Volume`` instances.
+
+# %%
+# Create an ``Image`` instance from random data.
+img_data = np.random.random((100, 100))
+img = Image(img_data)
+logging.info(f"img shape: {img.shape}")  # Note this produces a stack of 1.
```

Comment:
From what I recall, we need to output through the log system in order to please the linting gods. However, for a novice user, this will seem very odd. Is there no way we can carve out an exception for the tutorials (and let them use `print`)?

Reply:
Yes, we can, I do think in another gallery. I'll update this one.

Reply:
Great!